File: SensorConditionsTest/M30/LastStatus.java
package com.ichi2;


import org.javatuples.Pair;

import java.io.Serializable;
import java.util.ArrayList;

public class LastStatus implements Serializable {
    private ArrayList<Pair<String,Boolean>> allStatus;

    public LastStatus() {
        allStatus = new ArrayList<>();
    }

    public int amountStatusItens() {
        return allStatus.size();
    }

    public void addStatus(String sensorName, boolean status) {
        allStatus.add(new Pair<>(sensorName, status));
    }

    public boolean getStatus(String sensorName) {
        String sName;
        boolean status = false;

        for (Pair<String,Boolean> pair:allStatus) {
            sName = pair.getValue0();
            if (sName.equals(sensorName)) {
                status = pair.getValue1();
                break;
            }
        }

        return status;
    }

    public void setCurrentStatus(String sensorName, boolean status) {
        int index = 0;
        String sName;

        for (Pair<String,Boolean> pair:allStatus) {
            sName = pair.getValue0();
            if (sName.equals(sensorName)) {
                break;
            }
            index++;
        }
        allStatus.set(index, new Pair<>(sensorName,status));
    }

}

File: SensorConditionsTest/S10/LastStatus.kt
package com

import org.javatuples.Pair
import java.io.Serializable
import java.util.ArrayList

class LastStatus : Serializable {
    private val allStatus: ArrayList<Pair<String, Boolean>>
    fun amountStatusItens(): Int {
        return allStatus.size
    }

    fun addStatus(sensorName: String, status: Boolean) {
        allStatus.add(Pair(sensorName, status))
    }

    fun getStatus(sensorName: String): Boolean {
        var sName: String
        var status = false
        for (pair in allStatus) {
            sName = pair.value0
            if (sName == sensorName) {
                status = pair.value1
                break
            }
        }
        return status
    }

    fun setCurrentStatus(sensorName: String, status: Boolean) {
        var index = 0
        var sName: String
        for (pair in allStatus) {
            sName = pair.value0
            if (sName == sensorName) {
                break
            }
            index++
        }
        allStatus[index] = Pair(sensorName, status)
    }

    init {
        allStatus = ArrayList()
    }
}
